src/net.cpp

```diff
@@ -485,18 +485,27 @@ CNode* CConnman::ConnectNode(CAddress addrConnect, const char *pszDest, bool fCo
     Proxy proxy;
     CAddress addr_bind;
     assert(!addr_bind.IsValid());
+    std::unique_ptr<i2p::sam::Session> i2p_transient_session;
 
     if (addrConnect.IsValid()) {
+        const bool use_proxy{GetProxy(addrConnect.GetNetwork(), proxy)};
         bool proxyConnectionFailed = false;
 
-        if (addrConnect.GetNetwork() == NET_I2P && m_i2p_sam_session.get() != nullptr) {
+        if (addrConnect.GetNetwork() == NET_I2P && use_proxy) {
```

Comment:
The old codepath where `m_i2p_sam_session` exists  was not conditional on `use_proxy` before. Am I right that this is not a change in behavior, because for I2P, the existence of `m_i2p_sam_session` implies that use_proxy is true so it was just not necessary to check before?

Reply:
You are right. (`s/implies/implied/` in your comment above)

The code before this PR would set `m_i2p_sam_session` IFF there is a proxy for `NET_I2P`, so `m_i2p_sam_session.get() != nullptr` was equivalent to `GetProxy(NET_I2P, ...) == true` and checking just the former sufficed.

In the code after this PR we may have a situation where there is a proxy for `NET_I2P`, but `m_i2p_sam_session` is empty. Thus we check if a proxy is set and either use `m_i2p_sam_session` or make a transient session.